x-pack/plugins/security_solution/common/endpoint/service/authz/authz.ts

```diff
@@ -117,6 +117,9 @@ export const calculateEndpointAuthz = (
       hasEndpointManagementAccess,
       'readTrustedApplications'
     );
+
+  // TODO: ensure HIE privileges is used in the UI to hide/show security nav item
```

Comment:
What do you mean by this comment?

The check we do to determine if the menu option should be shown, I think, has to do with Downgrade scenarios - where a user might have defined some HIE, but then downgraded their license down. in that case, we check to see if data exists and allow the user to only delete entries if that is the case. thats the reason that the menu item for HIE is sometimes shown even if they don't have access to it.

That particular check - if data exists for HIE and determine if the user should see the menu item - should not be implemented here

Reply:
96d6fe2cc399aee2b68f9da5eb8bed9160183755